### PR TITLE
Fix kubectl autoscale dry run client side output

### PR DESCRIPTION
**Pull request by krishnaarava**
#### What type of PR is this?
/kind bug
/sig cli

#### Which issue(s) this PR fixes:
Fixes #[1060](https://github.com/kubernetes/kubectl/issues/1060)

#### Special notes for your reviewer:

**Before Fix:**

```
$ go run cmd/kubectl/kubectl.go autoscale --min=1 --max=4 Deployment/elasticsearch --namespace=foo --dry-run=client -o yaml
apiVersion: autoscaling/v1
kind: HorizontalPodAutoscaler
metadata:
  creationTimestamp: null
  name: elasticsearch
spec:
  maxReplicas: 4
  minReplicas: 1
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: elasticsearch
status:
  currentReplicas: 0
  desiredReplicas: 0
```

**After Fix:**
```
$ go run cmd/kubectl/kubectl.go autoscale --min=1 --max=4 Deployment/elasticsearch --namespace=foo --dry-run=client -o yaml
apiVersion: autoscaling/v1
kind: HorizontalPodAutoscaler
metadata:
  creationTimestamp: null
  name: elasticsearch
  namespace: foo
spec:
  maxReplicas: 4
  minReplicas: 1
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: elasticsearch
status:
  currentReplicas: 0
  desiredReplicas: 0
```

#### Does this PR introduce a user-facing change?
```release-note
Fix kubectl autoscale dry run client side output
```

### CODE REVIEW

**Comment by mehar** [comment]
Looks good to me

**Comment by lauchokyip** [comment]
`kubectl autoscale` does not ignore the `--namespace` flag, it will actually take the `--namespace` flag over here
https://github.com/kubernetes/kubernetes/blob/9f7c9c322fbfa1e92afeec0abdc64f8af2abe5e6/staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go#L166
and the namespace will be parsed to the server over here.
https://github.com/kubernetes/kubernetes/blob/9f7c9c322fbfa1e92afeec0abdc64f8af2abe5e6/staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go#L249

This PR will fix the `--dry-run=client-side` flag because when creating the Object, the namespace will not be parsed to the object. 
https://github.com/kubernetes/kubernetes/blob/9f7c9c322fbfa1e92afeec0abdc64f8af2abe5e6/staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go#L276-L288

I suggest changing the title to `Fix dry run client side output`

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go [hunk 1]
```diff
@@ -70,8 +70,8 @@ type AutoscaleOptions struct {
 
 	createAnnotation bool
 	args             []string
-	enforceNamespace bool
-	namespace        string
+	EnforceNamespace bool
```
Comment (lauchokyip) [inline staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go:73]: Is there any reasons why this is changed to capital letters?

**Comment by dims** [comment]
@krishnaarava there's a pending question from @lauchokyip PTAL

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go (hunk 1 above)
Comment (krishnaarava) [reply staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go:73]: Sorry missed this comment. Yeah need not be capital letters, let me update the PR.

**Pull request closed without merging**